### DISCUSSION TITLE
Console: `gen_random_uuid()` should be available for `uuid` fields like `now()` for timestamp fields when using mutations

### DISCUSSION THREAD

**Discussion by Zerebokep** [hasura/graphql-engine#8858] [General] [answered]
Generally it would be neat to use postgres functions inside queries and mutations.

**Comment by fonzarely** [comment]
you can already.
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/707217/182822461-faf3028c-323d-402d-9957-fb0fa83ab403.png">

**Comment by Zerebokep** [comment]
I mean using it in a query, not as default.

**Comment by plcplc** [comment, marked as answer]
As such, this is not a mechanism that Hasura provides, but rather it's Postgres itself which accepts that string as valid input text for the purpose of constructing a timestamp value (along with various [others](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE)).

So unfortunately there's no existing place we could extend to provide similar support for using `gen_random_uuid()` for uuid columns.

Postgres it seems doesn't let you overload the accepted textual input formats of a type, but it is possible to supply your own parsing function to custom types.

So in theory you should be able to define your own uuid type which supports `gen_random_uuid` in addition to literal uuids, although the practical value of doing this is questionable.
